mautic/mautic
# add pagehit index in entity metadata instead of fixture

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | []
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [x] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

When testing and working on the PHP 8 support (see https://github.com/mautic/mautic/pull/9969#issuecomment-1020322075), I noticed there are some fixtures that contain [DDL queries](https://www.geeksforgeeks.org/sql-ddl-dql-dml-dcl-tcl-commands/).

As Doctrine processes fixtures as 1 transaction that can be rolled back, and DDL queries force a commit when executed,
this results in an error when Doctrine tries to commit the transaction again.

in PHP < 8, this error was silently ignored, but in PHP >= 8 this error results in a ` There is no active transaction` PDOException.

One of the fixtures create an index on the `page_hits` table.
This was not possible to do in the `Hit` entity with existing functionality in the `ClassMetadataBuilder`, hence the workaround via a fixture.

This PR adds the extra functionality to support this and removed the fixture.
The extra functionality is also covered by a (minimal) test



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. install Mautic, and check if the `page_hits` table has an `page_hits_url` index




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10787"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
